File: apps/mybookkeeper/backend/app/schemas/applicants/screening_redirect_response.py
"""Response shape for ``GET /applicants/{id}/screening/redirect``.

PR 3.3 (rentals Phase 3, KeyCheck redirect-only): the backend looks up the
configured screening provider's dashboard URL and hands it to the frontend,
which then opens the URL in a new tab. The host completes the screening on
the provider's site and uploads the resulting PDF via the upload endpoint.
"""
from __future__ import annotations

from pydantic import BaseModel, ConfigDict


class ScreeningRedirectResponse(BaseModel):
    redirect_url: str
    provider: str

    model_config = ConfigDict(from_attributes=True)

File: apps/mybookkeeper/frontend/src/shared/types/screening/screening-redirect-response.ts
/**
 * Response shape for ``GET /applicants/{id}/screening/redirect``.
 *
 * The backend looks up the configured screening provider's dashboard URL
 * and the frontend opens it in a new tab. ``provider`` is included so the
 * UI can render "Open in KeyCheck" / "Open in <future provider>" without
 * a separate fetch.
 */
export interface ScreeningRedirectResponse {
  redirect_url: string;
  provider: string;
}
